### PR TITLE
LEA: Issue 154 - lea should use issuer from config

### LINKED ISSUE #154: LEA should use the issuer which was provided by the config
LEA is using a hard-coded String for its issuer to decode tokens. This is not correct. 

```
// opendut-lea/src/components/auth.rs

pub(crate) fn decode_token(token: &str) -> TokenData<Claims> {
    let mut validation = Validation::new(Algorithm::RS256);
    validation.set_issuer(&["https://keycloak/realms/opendut".to_string()]);  // TODO: get from config
    validation.set_audience(&["account".to_string()]);
    validation.insecure_disable_signature_validation();

    let decoding_key = DecodingKey::from_secret(&[]);

    jsonwebtoken::decode::<Claims>(token, &decoding_key, &validation).expect("failed to decode")
}
```

Instead of a hard-coded String the AppConfig offers the issuer as parameter, which can be used by LEA.

### PR DESCRIPTION
**Pull request by mtwardawski**
Fixes #154

### CODE REVIEW

File: opendut-lea/src/components/auth.rs [hunk 1]
```diff
@@ -58,6 +55,11 @@ pub fn LeaAuthenticated(
                     />
                 </Transition>
             }
+
+        }
+        _ => {
+            tracing::warn!("Warning: Authentication disabled. No issuer URL was provided by the config - not able to determine the authenticated user.");
```
Comment (reimarstier) [inline opendut-lea/src/components/auth.rs:61]: Add different warning log in case authentication config is present but not authenticated.

**Review by reimarstier — CHANGES_REQUESTED** [review]

File: opendut-lea/src/components/auth.rs [hunk 2]
```diff
@@ -58,6 +55,19 @@ pub fn LeaAuthenticated(
                     />
                 </Transition>
             }
+
+        }
+        (Some(lea_idp_config), None) => {
+            tracing::warn!("Warning: Authentication disabled - User not authenticated.");
```
Comment (reimarstier) [inline opendut-lea/src/components/auth.rs:61]: Isn't the authentication enabled when this config object is present?

**Review by reimarstier — CHANGES_REQUESTED** [review]

**Comment by reimarstier** [comment]
I noticed the CI did not complete successfully and judging from the output of the tests there are syntax errors present. 

Please address the following ToDos:
- [ ] Fix syntax errors
- [ ] Rebase your branch, there are potentially breaking changes due to #112
- [ ] Check if the test environment still works: 
  * `cargo ci distribution`
  * Re-run test environment
  * Check CARL's endpoint `https://carl/api/lea/config`
  * Make sure the configuration from the issuer contains a trailing slash (requirement by #112)
  * Check if authentication is pointing to correct address

File: opendut-lea/src/components/auth.rs (hunk 2 above)
Comment (mtwardawski) [reply opendut-lea/src/components/auth.rs]: That is right. Changed it.

**Pull request merged**